pkg/dataobj/planner/physical/dataobjscan.go

```diff
@@ -0,0 +1,37 @@
+package physical
+
+type DataObjLocation string
```

Comment:
I know it's going to be a pain in the ass, but what do you think about us getting in the habit of documenting any publicly exposed identifier? In my experience, it can really help with understanding the package API without needing to find references to see how something is used, especially when looking at it through pkg.go.dev.

Reply:
Generally, no objections. Think it's good habit. Some things are very obvious, though :D

Reply:
That's true, some things really may be obvious. But even if something is obvious, a doc comment can help solidify that "yes, this does what you think it does" and that there isn't anything special about it. 

For example, the doc comment for [bytes.Contains](https://pkg.go.dev/bytes@go1.24.1#Contains) confirms that "yes, this does exactly what you think it does," which doc comments for things like [math.Abs](https://pkg.go.dev/math#Abs) call out special cases to be aware of.

Reply:
Done. I think I covered everything